File: server/src/Models/DMToolkitModels/Definitions/SpeciesDefinition.cs
namespace DMToolkit.Models.Definitions;

public class SpeciesDefinition : IDefinition
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public int Speed { get; set; } = 0;
    public string Size { get; set; } = string.Empty;
    public ICollection<FeatDefinition> FeatDefinitions { get; set; } = new List<FeatDefinition>();
    public ICollection<LineageDefinition> LineageDefinitions = new List<LineageDefinition>();
}

File: web-client/src/lib/types/dm-tool-types/definitions/speciesDefinition.ts
import type { FeatDefinition } from "./featDefinition";
import type { LineageDefinition, LineageDefinitionDto } from "./lineageDefinition";

export interface SpeciesDefinition {
    id: string;
    name: string;
    description: string;
    type: string;
    speed: number;
    sizes: string[];
    featDefinitions: FeatDefinition[];
    lineageDefinitions: LineageDefinition[];
}

export interface SpeciesDefinitionDto {
    id: string;
    name: string;
    description: string;
    type: string;
    speed: number;
    sizes: string[];
    featDefinitionIds: string[];
    lineageDefinitions: LineageDefinitionDto[];
}